magni_bringup/scripts/launch_core.py

```
@@ -30,74 +35,197 @@
+                "pid_moving_buffer_size": str(motor_controller_params['pid_moving_buffer_size']),
+                "pid_velocity": str(motor_controller_params['pid_velocity']),
+                "wheel_separation_multiplier": str(vc_params['wheel_separation_multiplier']),
+                "wheel_radius_multiplier": str(vc_params['wheel_radius_multiplier']),
+                "has_velocity_limits": str(vc_lin['has_velocity_limits']),
+                "has_velocity_limits": str(vc_lin['max_velocity']),
+                "has_acceleration_limits": str(vc_lin['has_acceleration_limits']),
+                "max_acceleration": str(vc_lin['max_acceleration']),
+                "has_velocity_limits": str(vc_ang['has_velocity_limits']),
+                "max_velocity": str(vc_ang['max_velocity']),
+                "has_acceleration_limits": str(vc_ang['has_acceleration_limits']),
+                "max_acceleration": str(vc_ang['max_acceleration'])
+                }
+            )
+    except FileNotFoundError:
+        print("WARN "+em_path+" doesn't exist!")
+        return False
+    except (JSONDecodeError, NameError) as e:
+        print("WARN failed to parse "+em_path)
+        print(e)
+        return False        
+
+    f = open(path, "w")
+    f.write(expanded_em_launch)
+    f.close()
+    return True
+
+# finds file path by priority:
+# 1.) Checks first path and returns it if it exists
+# 2.) Checks second path and returns it if it exists
+# 3.) If none of them exists returns empty string
+def find_file_by_priority(first_path, second_path):
+    first_path = os.path.expanduser(first_path)
+    if not os.path.isfile(first_path):
+        # print("File "+first_path+" does not exsist")
+        second_path = os.path.expanduser(second_path)
+        if not os.path.isfile(second_path):
+            # print("File "+second_path+" does not exsist, using default config")
+            return ""
```

Should probably throw an exception that gets caught instead.

---

done